File: solutions/56-dipro/Age_of_villagers/Shape/Line.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Drawing;

namespace Age_of_villagers.Shape
{
    class Line : IShape 
    {
        private Point p1, p2;
        public Line(Point p1, Point p2)
        {
            this.p1 = p1;
            this.p2 = p2;
        }
        public void draw(Graphics g, Pen p)
        {
            g.DrawLine(p, p1, p2);
        }
    }
}

File: solutions/40-arif-JAVA/AgeOfVillagers/src/shape/Line.java
package shape;

import javafx.scene.canvas.GraphicsContext;

public class Line implements IShape {
    GraphicsContext graphicsContext;
    double x1,x2, y1,y2;
    public Line(GraphicsContext graphicsContext, double x1, double y1, double x2, double y2) {
        this.graphicsContext =graphicsContext;
        this.x1 =x1;
        this.x2 = x2;
        this.y1 = y1;
        this.y2 = y2;
    }

    @Override
    public void draw() {
        this.graphicsContext.strokeLine(x1,y1, x2,y2);
    }
}
